playwright-e2e/tests/dsm/miscellaneous/permissions/gp-collection-date-permissions-check.spec.ts

```diff
@@ -0,0 +1,111 @@
+        * Sent, Received, Sent/Received Overview, Search, Label Settings, Clinical Orders
+        */
+        const sampleMenu = new Dropdown(page, 'Samples');
+        const availableSampleMenuOptions = await sampleMenu.getDisplayedOptions<SamplesNavEnum>();
+        const amountOfSampleMenuOptions = availableSampleMenuOptions.length;
+        expect(
+          amountOfSampleMenuOptions,
+          `Expected 16 Sample menu options for a study admin with kit_shipping permission, received ${amountOfSampleMenuOptions} instead`).
+          toBe(16);
+        expect(availableSampleMenuOptions).toMatchObject(expectedSampleMenuItems);
+      })
+
+      await test.step('Go to Kits Sent page in the Samples menu and select a mf barcode and copy it', async () => {
+        await navigation.selectFromSamples(SamplesNavEnum.SENT);
+        const kitsSentPage = new KitsSentPage(page);
+        await kitsSentPage.waitForLoad();
+        await kitsSentPage.assertPageTitle();
+        await kitsSentPage.assertDisplayedKitTypes(expectedKitTypes);
+        await kitsSentPage.selectKitType(salivaKitType);
+        await kitsSentPage.rowsPerPage(50);
+
+        //Get the most recent mf barcodes (from within the last week) to be used in Kit Search page
+        await kitsSentPage.sortColumnByDate({ columnName: KitsColumnsEnum.SENT, startWithRecentDates: true });
+        const today = new Date();
+        const aWeekAgo = getDate(offsetDaysFromDate(today, 7, { isAdd: false }));
+        todaysKits = await kitsSentPage.getMFBarcodesSince(aWeekAgo, SamplesNavEnum.SENT);
+        for (const kits of todaysKits) {
+          const mfBarcode = (await kits.innerText()).trim();
+          todaysMFBarcodes.push(mfBarcode);
+        }
+        const numberOfRecentMFBarcodes = todaysMFBarcodes.length;
+        expect(numberOfRecentMFBarcodes, 'No mf barcodes from recent kits have been found').toBeGreaterThanOrEqual(1);
+      })
+
+      await test.step(`Go to Kit Search page and select 'Search by mf barcode' and enter the barcode you copied and click on Search`, async () => {
+        await navigation.selectFromSamples(SamplesNavEnum.SEARCH);
+        const kitsSearchPage = new SearchPage(page);
+        await kitsSearchPage.waitForReady();
+
+        //Find a kit that does not have a collection date and fill it out
+        for (let index = 0; index < todaysMFBarcodes.length; index++) {
+          const mfBarcode = todaysMFBarcodes[index];
+          logInfo(`MF Barcode: ${mfBarcode}`);
+          await kitsSearchPage.searchByField(SearchByField.MANUFACTURE_BARCODE, mfBarcode);
+          currentKit = page.locator('//app-field-datepicker//input');
+          const collectionDateField = await kitsSearchPage.getKitCollectionDate({ rowIndex: 1 });
+          logInfo(`Collection Date Field input: ${collectionDateField}`);
+          if (collectionDateField === '') {
+            break;
+          }
+          await page.reload();
```

Comment:
caution when using `page.reload`, ensure page is ready and loaded to avoid race condition flaky failures.

Reply:
will add a wait for ready